.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,44 @@
+# [install the GitHub Application on the relevant repositories](https://github.com/apps/pre-commit-ci-lite/installations/new)
+name: Run pre-commit.ci lite
+
+on:
+  pull_request:
+  push:
+    branches: [main, test-me-*]
+
+jobs:
+  main:
+    # note: the step must have either the default name or contain the text pre-commit-ci-lite. the application uses this to find the right workflow.
+    # [pre-commit.ci lite](https://pre-commit.ci/lite.html
+    name: pre-commit-ci-lite
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository and submodules
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.SGILE_PAT }}
+          submodules: recursive
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Install dependencies and package
+        run: |
+          pip install -r requirements.dev.txt
+          pip install pydantic
+      - name: flake8
+        run: |
+          flake8 --bug-report
+      - name: File changes
+        uses: trilom/file-changes-action@v1.2.4
+        id: file_changes
+        with:
+          prNumber: ${{ github.event.number }}
+          output: " "
+      - name: mypy
+        uses: pre-commit/action@v3.0.1
+        with:
```

Comment:
Why is mypy and flake8 added here too? Doesn't the precommit action run all of the precommit checks from the precommit config?

Reply:
I am guessing it's because you want to run black and isort on the whole codebase but only run mypy on the changes? We talked about this at the meeting though no? I thought we agreed to just run it all on the changes files since it is functionally equivalent and makes the ci more understandable/quicker